### ISSUE TITLE
Considering migrating to GitLab

### ISSUE THREAD

**Issue by rugk** [PrivateBin/PrivateBin#325]
So… we know why, I hardly have to explain it, I think.

So would you like that?
I saw that GitLab also has much more tools than GitHub, but of course the UI is thus a bit more complicated.

GitLab CE is at least FLOSS, so if GitLab.com is bought or fails for whatever reason, they would likely be forked or so if there is enough demand.

(Just up or downvote this issue here.)

## Community issues

* To any contributor: Would you like that? Would you move to GitLab, too, respectively create an account there?

## Technical issues

* [ ] Do the CI integrations work? If not, can we drop some – which one?
* [ ] anything else?

**Comment by r4sas** [comment]
What's the difference where to be? Choosing between Microsoft and Google is waste of time.
Anyway project is open, not in private repository.

So I think here no reason to move to GitLab.

**Comment by rugk** [comment]
> Choosing between Microsoft and Google is waste of time.

Gitlab is not owned by Google. It's owned by the GitLab B.V., i.e. an own company. Though one should know they have some venture capital from Google I think and host their servers at Azure.

As for why GitLab: Because there we have the way to self-host the project if it should be needed. We have a lower vendor lock-in as on GitHub. In any case it is better than Microsoft. This also addresses a big criticism of the [dear github](https://github.com/dear-github/dear-github/) letter: As FLOSS projects why should we actually host code on a closed-souce platform? We cannot even fix/add stuff if  
Also, features… I see they have a lot more, at least.

Self-hosting would be great, but hard work. We could also use a minor instance such as https://notabug.org/ (fork of gogs).

So I thought Gitlab is a great "middle-way" between self-hosting and staying on a large platform to have a good contributor base (GitLab seems to be [one of the big successors of the Microsoft deal](https://twitter.com/gitlab/status/1004777869862580224)). As you say, contributors are a big fact, that's why I've asked for this in this issue.
APIs for that CI stuff yes, that could be cumbersome. However, I've seen services with direct GitLab integration, but I think we don't use one currently. Also it's a thing you have to do once…

Finally, yes, this is totally low-prio right now and we can wait until GitHub experienced is ruined… :wink: (Though that may take some years – or does [this](https://twitter.com/UpEnd_org/status/1004292351643275264) already count – in case that story is true). But just to consider that possibility, I'd better ask now than doing it in a rush when it may be needed.

See also https://github.com/bderiso/Microsoft-Github-acquisition

**Comment by hasufell** [comment]
> maybe gogs

Yes, [gitea](https://gitea.io/) is currently the most active fork of gogs, which is very community oriented. Setting it up with docker is also relatively simple, almost as simple as PrivateBin! Since it supports OAuth it would not be an intrusive change for contributors. Maintaining a mirror on github or gitlab is great for availability.
